### PR TITLE
feat: support options

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
support options

#38534
#36152
#38856
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      1. Comments support 'options' parameter transfer 2. menu Repair using react18 antd_ 4.24.0, the menu flashes and the default style is lost. 3. Fix the menu flashes when the Sider of Layout is folded      |
| 🇨🇳 Chinese |      1 .  mentions支持`options`传参  2. menu 修复使用 react18 antd_4.24.0, 菜单闪烁，丢失默认样式 3.  修复Layout的Sider折叠时Menu闪烁  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
changelog 应包含3个minor的说明

File: components/mentions/index.zh-CN.md [hunk 1]
```diff
@@ -42,6 +54,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/jPE-itMFM/Mentions.svg
 | onResize | resize 回调 | function({ width, height }) | - |  |
 | onSearch | 搜索时触发 | (text: string, prefix: string) => void | - |  |
 | onSelect | 选择选项时触发 | (option: OptionProps, prefix: string) => void | - |  |
+| options | 选项配置 | [Options](#Option) | [] | 5.1.0 |
```
Comment (zombieJ) [inline components/mentions/index.zh-CN.md:57]: 版本错了~

Comment (heiyu4585) [reply components/mentions/index.zh-CN.md]: done

**Comment by heiyu4585** [comment]
> changelog 应包含3个minor的说明, 应提及 #38534 等...

没太理解,要怎么加

**Comment by yoyo837** [comment]
> 

我记得要求是`rc-*`升级要包含minor版本的变动说明.

**Comment by heiyu4585** [comment]
> 我记得要求是`rc-*`升级要包含minor版本的变动说明.

done

**Comment by afc163** [comment]
要改这里
图片

**Comment by heiyu4585** [comment]
> 要改这里 图片

done

File: components/mentions/index.zh-CN.md [hunk 2]
```diff
@@ -12,14 +12,26 @@ cover: https://gw.alipayobjects.com/zos/alicdn/jPE-itMFM/Mentions.svg
...
+import Alert from '../alert';
+ReactDOM.render(<Alert message="在 4.24.3 版本后，我们提供了 <Mentions options={[...]} /> 的简写方式，有更好的性能和更方便的数据组织方式，开发者不再需要自行拼接 JSX。同时我们废弃了原先的写法，你还是可以在 4.x 继续使用，但会在控制台看到警告，并会在 5.0 后移除。" />, mountNode);
+```
 
 ```jsx
+// >=4.24.3 可用，推荐的写法 ✅
+const options = [{ value: 'sample', label: 'sample' }];
+return <Mentions options={options} />;
+
+// <4.24.3 可用，>=4.24.3 时不推荐 🙅🏻‍♀️
```
Comment (yoyo837) [inline components/mentions/index.zh-CN.md:27]: 这不对, 4.24.3 依赖的 rc-mentions 还是 1.11.x, 所以新版后才能使用. 4.24.4? 4.25.0?

其他地方类似.

Comment (heiyu4585) [reply components/mentions/index.zh-CN.md:27]: 小于4.24.3的时候可以用,大于4.24.3( 4.24.4? 4.25.0)也可以用,不过不推荐使用了

Comment (yoyo837) [reply components/mentions/index.zh-CN.md:27]: 不是那个意思, 我是说`<Mentions options={options} />` 的写法似乎需要 `rc-mentions@1.12.0`, 但已发布的`antd@4.24.3`依赖的是1.11.x, 也就是 4.24.3 并不支持 `options`的用法.

Comment (yoyo837) [reply components/mentions/index.zh-CN.md:27]: 那么 `>=` 的描述就不正确

Comment (heiyu4585) [reply components/mentions/index.zh-CN.md]: 懂了,done

**Review by zombieJ — APPROVED** [review]

**Comment by heiyu4585** [comment]
merged

**Pull request merged**